enhancements/agent-installer/automated-workflow-kubeconfig.md

```diff
@@ -0,0 +1,253 @@
+  * Is depended on (directly or indirectly) by the implicating asset; or
+  * Itself implicates one of the assets that would be implicated if it could be
+    loaded from disk, but is not.
+
+then we must supply all assets that are implicated by the
+`kubeconfig.AdminClient` to the ignition generation step of the installer run
+by assisted-service. This will ensure that the built cluster accepts the
+credentials in the pre-existing kubeconfig.
+
+The status quo is that amongst the writable assets (shown in color below),
+there are some (shown in red) that cannot actually be read back by the
+installer. As a result, both the nature and the sheer number of assets required
+would preclude this approach:
+
+![Assets implicated by kubeconfig (status quo)](./kubeconfig_implications_without_changes.svg)
+
+However, by adding the ability to read back a handful of assets (shown in blue
+below) that currently cannot be, this complexity is massively reduced:
+
+![Assets implicated by kubeconfig with cert/key loading](./kubeconfig_implications_with_cert_loading.svg)
+
+Accordingly, the installer will be modified to allow reading the following
+certificate assets from disk:
+
+* `tls.KubeAPIServerLBSignerCertKey`
+* `tls.KubeAPIServerLocalhostSignerCertKey`
+* `tls.KubeAPIServerServiceNetworkSignerCertKey`
+* `tls.AdminKubeConfigSignerCertKey`
+* `tls.AdminKubeConfigClientCertKey`
+
+When generating the agent ISO, the certificate and key files for those same
+assets will be built in to the ISO.
+
+The assisted-service will copy the certificates from a location provided by the
+environment variable `ASSISTED_SERVICE_CLUSTER_TLS_CERTS_DIR` to the
+installer's work directory before generating the manifests/ignition.
+
+### Workflow Description
+
+```mermaid
+sequenceDiagram
+Cluster creator->>Agent installer: agent create image
+Agent installer->>Certificate assets: Generate()
+Agent installer->>Kubeconfig asset: Generate()
+Certificate assets-->>Kubeconfig asset: certificates
+Kubeconfig asset-->>Cluster creator: kubeconfig
+Agent installer->>Agent ISO: Generate()
+Certificate assets-->>Agent ISO: certificates
+Agent ISO-->>Cluster creator: agent.iso
+Agent ISO-->>Assisted Service: certificates
+Assisted Service-->>OpenShift Installer: certificates
+OpenShift Installer-->>Assisted Service: kubeconfig
+OpenShift Installer-->>Bootstrap: ignition
+```
+
+### API Extensions
+
+N/A
+
+### Implementation Details/Notes/Constraints
+
+Note that the modifications to the installer to read certificates from disk
+need to occur in the installer present in the payload of the release being
+installed, i.e. on the `master` branch, not the `agent-installer` branch where
+the agent-based installer is being developed.
+
+The work directory for the installer is a tmpdir created by assisted-service on
+the fly with a randomised name. Therefore it is not possible to pre-populate
+the certificates into the directory, and assisted-service must be involved in
+copying the files.
```

Comment:
Is that going to be the case for all manifests? Could we change the service to use either a well-defined location or a random one that both the service and agent can know?